### PR TITLE
Add a sigstore signing parameter file format, and CLI utility

### PR DESCRIPTION
**Pull request by mtrmac**
This defines a new file format (`containers-sigstore-signing-params.yaml(5)`), and provides an API to integrate it into callers.

That way, `skopeo` and `podman` can add a single `--sigstore-sign-by=`_param-file_ option, instead of 9 options; and users can maintain their site-specific configuration in one file, instead of manually copy&pasting 6 options in a typical case.

The parameter file format definition is a separate package that does not drag in the Rekor and Fulcio dependency tree; only actually _using_ the file has that cost.

I’m not too happy about the “containers sigstore signing parameters” name, nor about the `sigstore/params` subpackage name; I’ll certainly welcome suggestions.

(One possibility would be to make this a “containers signing parameters file”, not sigstore-specific, and to move all options in that file into a `sigstore` sub-object; then we could add a “simple signing” option to that same file as well.)

I also didn’t bother with using “functional options” for the `NewSignerFromParameterFile` function, nor with hiding the actual file format and only providing a constructor or an edit API. This is a non-essential utility and we can always provide a `cli/fulciov2` without impacting any existing callers, so keeping the API opaque and stable over time is not _that extremely_ important in this case.

Depends on #1785 .

@vrothberg PTAL. @rhatdan FYI.

### CODE REVIEW

**Comment by mtrmac** [comment]
Tested manually in Skopeo https://github.com/containers/skopeo/pull/1849 , basically with the provided example configs.

Adding unit tests is tracked in #1601.

**Comment by mtrmac** [comment]
Podman WIP: https://github.com/containers/common/pull/1288 https://github.com/containers/podman/pull/17088/files

**Review by vrothberg — APPROVED** [review]
LGTM

**Comment by mtrmac** [comment]
… and I do want to encourage changing the name of the file. I’m not happy at all with the current one.

**Comment by mtrmac** [comment]
Updated to quote YAML strings, motivated by https://ruudvanasseldonk.com/2023/01/11/the-yaml-document-from-hell .

(I chose to avoid TOML because of https://github.com/containers/image/issues/1054 , and YAML’s expression of nesting is much nicer. But I’m fine with any other format, to be consistent with the rest of the ecosystem.)

**Comment by vrothberg** [comment]
> Updated to quote YAML strings, motivated by https://ruudvanasseldonk.com/2023/01/11/the-yaml-document-from-hell .
> 
> (I chose to avoid TOML because of #1054 , and YAML’s expression of nesting is much nicer. But I’m fine with any other format, to be consistent with the rest of the ecosystem.)

LGTM

I don't have an opinion on the file format.  Made me think of Mutt's slog: "All mail clients suck. This one just sucks less."

File: docs/containers-sigstore-signing-params.yaml.5.md
Comment (TomSweeneyRedHat) [inline docs/containers-sigstore-signing-params.yaml.5.md:8]:
```suggestion
# DESCRIPTION
```

File: docs/containers-sigstore-signing-params.yaml.5.md
Comment (TomSweeneyRedHat) [inline docs/containers-sigstore-signing-params.yaml.5.md:10]: I thought this was going to be a single file?  If so, suggest:
```suggestion
The sigstore signing parameter file is  used to store options that may be required to create sigstore signatures.
```

Comment (TomSweeneyRedHat) [reply docs/containers-sigstore-signing-params.yaml.5.md:10]: Ignore, I see where you were getting at down below.

File: docs/containers-sigstore-signing-params.yaml.5.md
Comment (TomSweeneyRedHat) [inline docs/containers-sigstore-signing-params.yaml.5.md:11]: On the fence, your call.  "standard location" or "default location"?

Comment (mtrmac) [reply docs/containers-sigstore-signing-params.yaml.5.md:11]: (I’m not very sure that sentence even needs to exist… I just wanted to give _some_ context for users who stumble on the page directly, without starting at a `--sign-by-sigstore` option documentation.)

Sure, “default location” works.

File: docs/containers-sigstore-signing-params.yaml.5.md
Comment (TomSweeneyRedHat) [inline docs/containers-sigstore-signing-params.yaml.5.md:57]: Based solely on grokking Google, it looks like it should be "OpenID Connect".  (https://openid.net/connect/).  Please change here and elsewhere.  I've added quotes to set it off from the rest of the text and to let the reader know that's a thing that could be looked up if they're unfamiliar as I was.  That said, I'm still on the fence about having the quotes, so no foul if you want to drop them.

```suggestion
  `staticToken` provides a pre-existing "OpenID Connect" ID token which must have been obtained separately.
```

Comment (TomSweeneyRedHat) [reply docs/containers-sigstore-signing-params.yaml.5.md:57]: I'll hop off the fence, drop the quotes.  I see you have a number of instances and it would look odd all over.

Comment (mtrmac) [reply docs/containers-sigstore-signing-params.yaml.5.md]: Thanks! The quotes are a great idea for separating the two terms.

(I didn’t find any other instance of “Open ID” with a space.)

**Comment by TomSweeneyRedHat** [comment]
Couple doc nits which I think @mtrmac has tended to already, otherwise, LGTM.

File: docs/containers-sigstore-signing-params.yaml.5.md
Comment (mtrmac) [reply docs/containers-sigstore-signing-params.yaml.5.md]: How about quoting the “ID token” part instead? That would be less useful as a stand-alone search term, but assuming the readers figure out `OpenID Connect` is a term to search from other parts of the page, making it clear “ID token” is a single phrase might be worthwhile.

**Comment by mtrmac** [comment]
@TomSweeneyRedHat Thanks a lot!

File: docs/containers-sigstore-signing-params.yaml.5.md
Comment (TomSweeneyRedHat) [reply docs/containers-sigstore-signing-params.yaml.5.md]: Works for me!

Comment (mtrmac) [reply docs/containers-sigstore-signing-params.yaml.5.md]: Thanks, changed to quote “ID token”.

**Review by TomSweeneyRedHat — COMMENTED** [review]
LGTM

**Pull request merged**
